### PR TITLE
REM-294/ca-opening-balances

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
ca-opening-balances
Cash And Banks / Activities / Opening Balances

### CODE REVIEW

File: src/repositories/CashBankRepository.js [hunk 1]
```diff
@@ -52,5 +52,12 @@ export const CashBankRepository = {
   AccountBalance: {
     qry: service + 'qryABA',
     get: service + 'getABA'
+  },
+  OpeningBalance: {
+    page: service + 'qryOBA',
```
Comment (FatimaSalami) [inline src/repositories/CashBankRepository.js:57]: this is qry

File: src/pages/ca-opening-balances/form/OpeningBalanceForm.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      amount: yup.number().required(' ')
+    }),
+    onSubmit: async obj => {
+      const currencyId = formik.values.currencyId
```
Comment (Ghinwa-Arab) [inline src/pages/ca-opening-balances/form/OpeningBalanceForm.js:47]: Can we use those 3 fields directly without initializing them before?

File: src/pages/ca-opening-balances/form/OpeningBalanceForm.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      const fiscalYear = formik.values.fiscalYear
+      const accountId = formik.values.accountId
+
+      const response = await postRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/ca-opening-balances/form/OpeningBalanceForm.js:51]: if response not used below directly do
await postRequest({
        extension: CashBankRepository.OpeningBalance.set,
        record: JSON.stringify(obj)
      })

Comment (Alinadimmokdad) [reply src/pages/ca-opening-balances/form/OpeningBalanceForm.js:51]: done

File: src/pages/ca-opening-balances/form/OpeningBalanceForm.js (hunk 2 above)
Comment (Alinadimmokdad) [reply src/pages/ca-opening-balances/form/OpeningBalanceForm.js:47]: the if statement should work after the value of them

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**